### DISCUSSION TITLE
How can I find list of properties that I can set by "SetProp"?

### DISCUSSION THREAD

**Discussion by mohammad-saber** [rdkit/rdkit#4043] [Q&A]
Hi everyone!

As you know there is a method that we can set properties of molecules, atoms and, I think, bonds like below:

`atom.SetProp("_displayLabel",atom.GetSymbol())`

I would like to know where I can find the list of properties that we can set by this method. Especially, I am looking for the following properties:

1) Color for a specific atom in the molecule (I can change the color for the desired atom)

2) Shape of the bond between two atoms. For example, I change the bond from a single bond into an aromatic bond. By aromatic bond, I mean a bond that looks like below:

<img width="81" alt="Screen Shot 2021-04-16 at 9 58 11" src="https://user-images.githubusercontent.com/38423900/114956783-8b0d7800-9e9a-11eb-8c93-49b59d6471e0.png">

**Comment by greglandrum** [comment]
You can set whatever properties you want with `SetProp()`. I guess you're asking which ones have special meanings.
There's a partial list of those here: https://www.rdkit.org/docs/RDKit_Book.html#magic-property-values

To answer the specific questions you ask:
1. Changing the color of an atom in the molecule: assuming you are asking about changing the color used to draw the atom symbol, you can't directly do this since the drawing color is determined by the atomic number. You can update the palette for those colors though. Here's an example of to do that: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/MolDraw2D/Wrap/testMolDraw2D.py#L271  If you want to highlight some atoms or bonds you can tell the RDKit to do that: https://www.rdkit.org/docs/GettingStartedInPython.html#drawing-molecules
2. Changing the way a bond is drawn. I don't think I understand this one. If you want the bond to be drawn like an aromatic bond, then set the bond type to aromatic. If you want it drawn like a single bond set the bond type to single.

**Reply by mohammad-saber** [reply]
Thank you for your help. 

Purpose of asking this question:

I am trying to draw "C(C(N)=O)c(c)c". But I get the following error. 

`RDKit ERROR: [11:17:55] non-ring atom 4 marked aromatic
`

For visualization purpose, I convert "c" into "C". So, I can draw "C(C(N)=O)C(C)C" as below:

<img width="350" alt="Screen Shot 2021-04-17 at 13 12 32" src="https://user-images.githubusercontent.com/38423900/115101377-a2be2c80-9f7e-11eb-8de5-e459faaf3c1c.png">


But I need to show that several carbon atoms are in fact aromatic carbons. One idea is changing the shape of bonds between these atoms into aromatic bonds. If it is not possible, I try to use another color for these atoms.

**Reply by greglandrum** [reply]
It could be that you are over-complicating things.
You could just parse the SMILES with partial sanitization and leave the aromatic flags intact.
You need to do a bit more work to make this happen, and you need to tell the drawing code not to kekulize the bonds before drawing, but that's not hard:
https://gist.github.com/greglandrum/73c488981caa83da66fd4b08b219a64a

**Comment by mohammad-saber** [comment]
Following my last comment, what I am trying to draw is similar to Morgan bits or RDK bits as below:

![s](https://user-images.githubusercontent.com/38423900/115172900-1ee07d80-a101-11eb-917e-b3d75327cd8a.png)


As can be seen, the aromatic bonds are specified very clearly. I try to visualize my desired SMILE similarly. However, I couldn't find the method or function to do that. 

If you have any idea, I really appreciate it to guide me on this issue.

**Reply by greglandrum** [reply]
Probably the best way to learn how to draw things that way is to just look at the code:
https://github.com/rdkit/rdkit/blob/master/rdkit/Chem/Draw/__init__.py#L983
